valhalla/thor/pathalgorithm.h

```diff
@@ -119,9 +142,51 @@ class PathAlgorithm {
     const baldr::GraphTile* tile = graphreader.GetGraphTile(node);
     return (tile == nullptr) ? 0 : tile->node(node)->timezone();
   }
+
+  /**
+   * Adds an edge to the expansion history of the current pass of the algorithm
+   * There is no schema for the status, but loosely you should pass one of:
+   * reached, settled, connected so the person using this geojson can style accordingly
+   */
+  virtual inline void TrackExpansion(baldr::GraphReader& reader,
+                                     baldr::GraphId edgeid,
+                                     const char* status,
+                                     bool full_shape = false) {
+    // only do this if someone is looking for this on the outside
+    if (auto expansion = expansion_.lock()) {
```

Comment:
this is where we check we should actually be tracking this information or not for this run of the path algorithms getbestpath function